### ISSUE TITLE
Rule proposal: force-boolean-literal-compare

### ISSUE THREAD

**Issue by brunosabot** [typescript-eslint/typescript-eslint#3974]
### Description

This rule is aimed to do the reverse of `no-unnecessary-boolean-literal-compare`.
In our base code, we prefer to have a literal compare instead of a negate conditon for a better visibility.

```ts
// The negation is not very readable
if (!myCondition) {
  // do something
}
```

```ts
// The negation is clearer
if (myCondition === false) {
  // do something
}
```

Ideally, the would have a parameter _shouldForceTruthCompare_ to also activate it for the truth condition, but this one should be optional.

### Fail

```ts
if (!myCondition) {
  // do something
}
```
With the option for _shouldForceTruthCompare_ parameter:
```ts
if (myCondition) {
  // do something
}
```

### Pass

```ts
if (myCondition === false) {
  // do something
}
```
With the option for _shouldForceTruthCompare_ parameter:
```ts
if (myCondition === true) {
  // do something
}
```
